src/main/java/groovy/lang/MetaClassImpl.java

```diff
@@ -2207,11 +2209,28 @@ private static MetaMethod getCategoryMethodSetter(Class sender, String name, boo
 //                    element = new MetaBeanProperty(mp.getName(), mp.getType(), null, mp.getSetter());
 //                }
             }
+            if (element instanceof MetaBeanProperty) {
+                MetaBeanProperty mbp = (MetaBeanProperty) element;
+                boolean getterAccessible = canAccess(mbp.getGetter());
+                boolean setterAccessible = canAccess(mbp.getSetter());
+
+                if (!(getterAccessible && setterAccessible)) continue;
```

Comment:
@paulk-asert Could you review the logic of fitering? 
getter and setter of property should be access legally, or the property will be filtered for now.
Also, filtering non-properties is a breaking change. Should we provide an option to allow the illegal access?(Suggestion: forbiddening illegal access by default)